java/src/jmri/jmrix/lenz/XNetTurnout.java

```diff
@@ -435,41 +421,26 @@ private synchronized void handleMonitoringModeFeedback(XNetReply l) {
          */
         log.debug("Handle Message for turnout {} in MONITORING feedback mode ", mNumber);
         if (internalState == IDLE || internalState == STATUSREQUESTSENT) {
-            if (l.isFeedbackBroadcastMessage()) {
-                // This is a feedback message, we need to check and see if it
-                // indicates this turnout is to change state or if it is for
-                // another turnout.
-                int numDataBytes = l.getElement(0) & 0x0f;
-                for (int i = 1; i < numDataBytes; i += 2) {
-                    if (parseFeedbackMessage(l, i) != -1) {
-                        log.debug("Turnout {} MONITORING feedback mode - state change from feedback.", mNumber);
-                        break;
-                    }
-                }
+            if (l.onTurnoutFeedback(mNumber, this::parseFeedbackMessage)) {
+                log.debug("Turnout {} MONITORING feedback mode - state change from feedback.", mNumber);
             }
         } else if (getCommandedState() != getKnownState()
                 || internalState == COMMANDSENT) {
-            if (l.isFeedbackBroadcastMessage()) {
-                int numDataBytes = l.getElement(0) & 0x0f;
-                for (int i = 1; i < numDataBytes; i += 2) {
-                    int messageType = l.getFeedbackMessageType(i);
-                    if (messageType == 0 || messageType == 1) {
-                        // In Monitoring mode, treat both turnouts with feedback
-                        // and turnouts without feedback as turnouts without
-                        // feedback.  i.e. just interpret the feedback
-                        // message, don't check to see if the motion is complete
-                        if (parseFeedbackMessage(l, i) != -1) {
-                            // We need to tell the turnout to shut off the output.
-                            log.debug("Turnout {} MONITORING feedback mode - state change from feedback, CommandedState != KnownState.", mNumber);
-                            sendOffMessage();
```

Comment:
Removing this may be problematic.
The feedback has to be able to trigger an off message.
I will need to verify this works with real hardware.